File: tour_back/src/main/java/dev/tin/tour_back/dto/TourDTO.java
package dev.tin.tour_back.dto;

import dev.tin.tour_back.entity.ItineraryEntity;
import dev.tin.tour_back.entity.TourPriceEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TourDTO {
    private String name;
    private String description;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private Integer maxQuantity;
    private String approvalStatus;
    private List<TourPriceEntity> tourPrices;
    private List<Long> typeOfTourIds;
    private List<ItineraryEntity> itineraries;
    private List<String> itineraryStrings;
    private List<ServiceDTO> services;
    // Control flags
    private boolean clearAllPrices;
    private boolean clearAllServices;
} 

File: tour_front/src/dto/TourDTO.js
/**
 * Data Transfer Object for Tour
 * Ensures tour data is formatted correctly for API
 */
class TourDTO {
  constructor(data = {}) {
    // Basic information
    this.id = data.id;
    this.name = data.name || '';
    this.description = data.description || '';
    this.maxQuantity = data.maxQuantity ? parseInt(data.maxQuantity) : 0;
    this.approvalStatus = data.approvalStatus || 'PENDING';
    
    // Dates
    this.startDate = data.startDate || null;
    this.endDate = data.endDate || null;
    
    // Related objects
    this.typeOfTourIds = data.typeOfTourIds || [];
    this.tourPrices = data.tourPrices || [];
    this.services = data.services || [];
    this.itineraries = data.itineraries || [];
    this.itineraryStrings = data.itineraryStrings || [];
    
    // Control flags for API
    this.clearAllPrices = true;
    this.clearAllServices = true;
  }
}

export default TourDTO; 